File: Others/easy/TournamentWinner.py
def tournamentWinner(competitions, results):

    scores = {}

    for i in range(len(results)):
        winner = competitions[i][0] if results[i] == 1 else competitions[i][1]
        current_score = scores.get(winner, 0)
        scores[winner] = current_score + 3

    max_score = max(scores.values())
    for key, value in scores.items():
        if value == max_score:
            return key

File: Golang/easy/tournament-winner.go
package main

func TournamentWinner(competitions [][]string, results []int) string {
	scores := make(map[string]int)
	bestScore := 0
	bestLang := ""
	for idx, competition := range competitions {
		key := competition[1-results[idx]]
		scores[key] += 3
		if scores[key] > bestScore {
			bestScore = scores[key]
			bestLang = key
		}
	}
	return bestLang
}
